src/main/java/com/google/cloud/tools/maven/DeployMojo.java

```diff
@@ -0,0 +1,157 @@
+package com.google.cloud.tools.maven;
+
+/**
+ * Stages and then deploys application to App Engine standard or flexible environment.
+ */
+
+import com.google.cloud.tools.app.api.deploy.AppEngineDeployment;
+import com.google.cloud.tools.app.api.deploy.DeployConfiguration;
+import com.google.cloud.tools.app.impl.cloudsdk.CloudSdkAppEngineDeployment;
+
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.Execute;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+
+import java.io.File;
+import java.util.List;
+
+/**
+ * Stage and deploy an application to Google App Engine standard or flexible environment.
+ */
+@Mojo(name = "deploy")
+@Execute(phase = LifecyclePhase.PACKAGE)
+public class DeployMojo extends StageMojo implements DeployConfiguration {
+
+  /**
+   * The yaml files for the services or configurations you want to deploy. If not given, defaults to
+   * app.yaml in the current directory. If that is not found, attempts to automatically generate
+   * necessary configuration files (such as app.yaml) in the current directory.
+   */
+  @Parameter(alias = "deploy.deployables", property = "app.deploy.deployables")
```

Comment:
It turns out `alias` is what needs to be used. `name` is used when you want to use a getter for the property. The documentation on this is not super clear, but I found this out by experimenting.